src/main/java/com/hcl/appscan/jenkins/plugin/scanners/StaticAnalyzer.java

```diff
@@ -77,29 +90,84 @@ public String checkScanSpeed(String scanSpeed) {
         		}
         	return null;
     	}
-       
-        public boolean isOpenSourceOnly() {
-            if(!m_scanMethod.equals(CoreConstants.UPLOAD_DIRECT)){
-                return m_openSourceOnly;
-            }
-            return false;
-        }
-        
+
         @DataBoundSetter
         public void setOpenSourceOnly(boolean openSourceOnly) {
             m_openSourceOnly = openSourceOnly;
         }
 
-        public boolean isSourceCodeOnly() {
+        public boolean getOpenSourceOnly() {
+            return m_openSourceOnly;
+        }
+        
+        @DataBoundSetter
+        public void setIncludeSCAGenerateIRX(String includeSCAGenerateIRX) {
+            if(getHasOptions() && includeSCAGenerateIRX.equals("true")) {
+                m_includeSCAGenerateIRX = "true";
+            } else {
+                m_includeSCAGenerateIRX = "false";
+            }
+        }
+
+        public String getIncludeSCAGenerateIRX() {
             if(!m_scanMethod.equals(CoreConstants.UPLOAD_DIRECT)){
-                return m_sourceCodeOnly;
+                return m_includeSCAGenerateIRX;
             }
-            return false;
+            return "";
+        }
+
+        //using this method in the jelly file to determine the checkbox value
+        public String isIncludeSCAGenerateIRX(String includeSCAGenerateIRX) {
+            if (m_includeSCAGenerateIRX != null) {
+                return m_includeSCAGenerateIRX.equalsIgnoreCase(includeSCAGenerateIRX) ? "true" : "false";
+            }
+            return "true";
+        }
+
+        @DataBoundSetter
+        public void setHasOptionsUploadDirect(boolean hasOptionsUploadDirect) {
+            m_hasOptionsUploadDirect = hasOptionsUploadDirect;
+        }
+
+        //using this method in the jelly file to determine the checkbox value
+        public boolean getHasOptionsUploadDirect() {
+            return m_hasOptionsUploadDirect;
+        }
+
+        @DataBoundSetter
+        public void setIncludeSCAUploadDirect(String includeSCAUploadDirect) {
+            if(m_hasOptionsUploadDirect && includeSCAUploadDirect.equals("true")) {
```

Comment:
Should use `Boolean.parseBoolean(string)`.

Reply:
Okay, sure.